feature-libs/cart/base/components/clear-cart/clear-cart-dialog/clear-cart-dialog-component.service.ts

```diff
@@ -1,77 +1,82 @@
 import { Injectable } from '@angular/core';
-import { GlobalMessageService, GlobalMessageType } from '@spartacus/core';
-import { ActiveCartFacade } from '@spartacus/cart/base/root';
-import { filter, switchMap, take, tap } from 'rxjs/operators';
-import { BehaviorSubject, Observable } from 'rxjs';
+import {
+  EventService,
+  GlobalMessageService,
+  GlobalMessageType,
+  UserIdService,
+} from '@spartacus/core';
+import {
+  ActiveCartFacade,
+  MultiCartFacade,
+  DeleteCartSuccessEvent,
+  DeleteCartFailEvent,
+} from '@spartacus/cart/base/root';
+import { mapTo, switchMap, take, tap, withLatestFrom } from 'rxjs/operators';
 import { LaunchDialogService } from '@spartacus/storefront';
+import { merge } from 'rxjs';
 
 @Injectable({
   providedIn: 'root',
 })
 export class ClearCartDialogComponentService {
-  isClearing$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
-
   constructor(
     protected launchDialogService: LaunchDialogService,
     protected globalMessageService: GlobalMessageService,
-    protected activeCartFacade: ActiveCartFacade
+    protected activeCartFacade: ActiveCartFacade,
+    protected multiCartFacade: MultiCartFacade,
+    protected userIdService: UserIdService,
+    protected eventService: EventService
   ) {}
 
+  /**
+   * Clear all entries and info of active cart
+   */
   clearActiveCart(): void {
-    this.isClearing$.next(true);
-    this.clearCart()
-      .pipe(take(1))
-      .subscribe(() => {
-        this.isClearing$.next(false);
-        this.onComplete();
+    this.activeCartFacade
+      .getActiveCartId()
+      .pipe(
+        withLatestFrom(this.userIdService.getUserId()),
+        take(1),
+        tap(([cartId, userId]) => {
+          this.multiCartFacade.deleteCart(cartId, userId);
+        }),
+        switchMap(() =>
+          merge(
+            this.eventService.get(DeleteCartSuccessEvent).pipe(mapTo(true)),
+            this.eventService.get(DeleteCartFailEvent).pipe(mapTo(false))
+          )
+        ),
+        tap(() => this.closeDialog('Close dialog after cart cleared')),
+        take(1)
```

Comment:
Ah! Nice use of the `merge` operator for catching 2 events.
Perhaps move the  take(1) on line 50 to `.pipe(take(1)` directly on the merge statement.

Reply:
There is no unsubscribe, are the take(1) statements enough to complete the stream in spite of the switchMap ?